File: Ex1.java
import java.util.ArrayList;
import java.util.List;

public class Main {

    static List<String> findSubstrings(String str) {
        List<String> substrings = new ArrayList<>();

        for (int i = 0; i < str.length(); ++i) {
            if (str.charAt(i) == '1' && i < str.length() - 2 && str.charAt(i + 1) == '0') {
                int j = i + 2;
                while (j < str.length() && str.charAt(j) == '0') {
                    ++j;
                }
                if (j < str.length() && str.charAt(j) == '1') {
                    substrings.add(str.substring(i, j + 1));
                }
                i = j - 1;
            }
        }

        return substrings;
    }

    public static void main(String[] args) {
        String str1 = "101tf11flb10001";
        String str2 = "101001tf11flb10001";

        List<String> result1 = findSubstrings(str1);
        List<String> result2 = findSubstrings(str2);

        System.out.println("Результат для первой строки: " + String.join(", ", result1) + ".");
        System.out.println("Результат для второй строки: " + String.join(", ", result2) + ".");
    }
}

File: Ex1.kt
fun findSubstrings(str: String): List<String> {
    val substrings = mutableListOf<String>()
    var i = 0
    while (i < str.length) {
        if (str[i] == '1' && i < str.length - 2 && str[i + 1] == '0') {
            var j = i + 2
            while (j < str.length && str[j] == '0') {
                j++
            }
            if (j < str.length && str[j] == '1') {
                substrings.add(str.substring(i, j + 1))
            }
            i = j - 1
        }
        i++
    }
    return substrings
}

fun main() {
    val str1 = "101tf11flb10001"
    val str2 = "101001tf11flb10001"

    val result1 = findSubstrings(str1)
    val result2 = findSubstrings(str2)

    println("Результат для первой строки: ${result1.joinToString(", ")}.")
    println("Результат для второй строки: ${result2.joinToString(", ")}.")
}
